src/coreclr/jit/hwintrinsiclistxarch.h

```diff
@@ -713,6 +722,21 @@ HARDWARE_INTRINSIC(AVX2,            SubtractSaturate,
 HARDWARE_INTRINSIC(AVX2,            Xor,                                        32,              2,     {INS_pxor,              INS_pxor,               INS_pxor,               INS_pxor,               INS_pxor,               INS_pxor,               INS_pxor,               INS_pxor,               INS_invalid,            INS_invalid},           HW_Category_SimpleSIMD,             HW_Flag_Commutative)
+// ***************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//                 ISA              Function name                               SIMD size       NumArg                                                                                                         Instructions                                                                                                                             Category                            Flags
+//                                                                                                      {TYP_BYTE,              TYP_UBYTE,              TYP_SHORT,              TYP_USHORT,             TYP_INT,                TYP_UINT,               TYP_LONG,               TYP_ULONG,              TYP_FLOAT,              TYP_DOUBLE}
+// ***************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//  AVX512F Intrinsics
+HARDWARE_INTRINSIC(AVX512F,         BroadcastScalarToVector512,                 64,              1,     {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_vpbroadcastd,       INS_vpbroadcastd,       INS_vpbroadcastq,       INS_vpbroadcastq,       INS_vbroadcastss,       INS_vbroadcastsd},      HW_Category_SIMDScalar,             HW_Flag_MaybeMemoryLoad)
+HARDWARE_INTRINSIC(AVX512F,         InsertVector256,                            64,              3,     {INS_vinserti64x4,      INS_vinserti64x4,       INS_vinserti64x4,       INS_vinserti64x4,       INS_vinserti64x4,       INS_vinserti64x4,       INS_vinserti64x4,       INS_vinserti64x4,       INS_vinsertf64x4,       INS_vinsertf64x4},      HW_Category_IMM,                    HW_Flag_FullRangeIMM)
```

Comment:
Shouldn't `float` be `vinsertf32x8` and likewise `vinserti32x8` for `int`/`uint`?

There isn't a "specific instruction" for byte/sbyte or short/ushort, so we can really just use whatever there.

Reply:
Yeah. If we are checking for `DQ` too, then makes sense to switch. Functionally I was not sure if there is a difference here `vinserti32x8`

Reply:
I don't believe there is when masking isn't involved. It is then much like `vmovdqu8` vs `vmovdqu16` in that when no kmask exists, either can be used, but using the "right" one can help with reading disassembly/etc.

Reply:
Ah...actually `vinserti32x8`   `vinsertf32x8` and can only be used for `AVX512DQ.InsertVector256` :https://www.felixcloutier.com/x86/vinsertf128:vinsertf32x4:vinsertf64x2:vinsertf32x8:vinsertf64x4

So, for AVx512F, it'll always have to be `i64x4`. That's how I understand the AVX512F vs AVx512DQ API for c#. Feel free to correct me here

Reply:
Ah, yeah. I missed that they were `AVX512DQ` only. Makes sense then.